test/models/location_test.rb

```diff
@@ -1,6 +1,62 @@
 require 'test_helper'
 require 'models/shared/taxonomies_base_test'
+require 'rfauxfactory'
 
 class LocationTest < ActiveSupport::TestCase
   include TaxonomiesBaseTest
+  valid_loc_data_list = [
+    RFauxFactory.gen_alpha(1),
+    RFauxFactory.gen_alpha(246),
+    *RFauxFactory.gen_strings(1..246, exclude: [:html]).values,
+    RFauxFactory.gen_html(rand((1..221)))
+  ]
+  invalid_loc_name_list = [
+    '',
+    ' ',
+    "\t",
+    *RFauxFactory.gen_strings(247).values
+  ]
+
+  test 'should create with multiple valid names' do
+    valid_loc_data_list.each do |name|
+      location = FactoryBot.create(:location, :name => name)
+      location.reload
+      assert_equal name, location.name
+    end
+  end
+
+  test 'should not create with multiple invalid names' do
+    invalid_loc_name_list.each do |name|
+      location = Location.new(:name => name)
```

Comment:
it would be better to use FactoryBot.build(:location, :name => name) since that ensures that the new loc object is valid (except for the invalid name). If loc would have other attributes which presence is required, or other name validations, the test could pass without catching the name validations we're testing.